cmd/execution/main.go

```diff
@@ -385,6 +389,8 @@ func main() {
 				chdpDeliveryTimeout,
 			)
 
+			logEpochNumber(vm, vmCtx, executionState, node, node.Logger.With().Str("component", "provider engine").Logger())
```

Comment:
I would suggest to organize the logic a bit, and we should crash the node if we ran into any error, or in the end the epoch counter in protocol state and smart contract is mismatching.

```golang
epochCounterFromProtocolState, err := node.State.Final().Epochs().Current().Counter()
if err != nil {
  return fmt.Errorf(...) // we would like to stop the node if we fail to query protocol state
}
epochCounterFromSmartContract, err := readEpochCounterFromSmartContract(...)
if err != nil {
  return fmt.Errorf(...) // we would like to stop the node if we fail to query smart contract
}
// we would like to stop the node if the epoch counter is mismatching
if epochCounterFromProtocolState != epochCounterFromSmartContract {
  return fmt.Errorf("mismatching epoch counter between protocol state %v and smart contract %v", 
  epochCounterFromProtocolState, epochCounterFromSmartContract)
} 

log.Info().Uint64("epochCounter", epochCounterFromProtocolState).Msg("epoch counter is matching")
```

Reply:
> we should crash the node if we ran into any error, or in the end the epoch counter in protocol state and smart contract is mismatching

@zhangchiqing We can't do this because immediately following a spork they will be mismatching until the `resetEpoch` transaction is submitted.

Reply:
I see. Right. Let's simply logging them then.